### ISSUE TITLE
`softmax` returns `NaN` when inputs are `inf`

### ISSUE THREAD

**Issue by jakevdp** [google/jax#21495]
Example:
```python
In [1]: import jax

In [2]: import jax.numpy as jnp

In [3]: jax.nn.softmax(jnp.array([jnp.inf, 0]))
Out[3]: Array([nan, nan], dtype=float32)
```
This should probably return `Array([1.,  0.)` instead.

With multiple infinities, the output should in theory depend on the cardinality of the infinities: i.e. `jax.nn.softmax(jnp.array([jnp.inf, 0, jnp.inf]))` should probably output `Array([0.5, 0. , 0.5])`.

**Comment by jakevdp** [comment]
For what it's worth, both `scipy` and `pytorch` return `NaN` in this case:
```python
from scipy import special
import numpy as np
x = np.array([np.inf, 0, np.inf])
special.softmax(x)
# array([nan, nan, nan])

from torch import nn
import torch
m = nn.Softmax(dim=0)
input = torch.from_numpy(x)
m(input)
# tensor([nan, nan, nan], dtype=torch.float64)
```

**Comment by jakevdp** [comment]
After some offline discussion, I think the best resolution here will be to document that the output will be NaN for infinite inputs. The value of `inf / inf` or `inf - inf` is undefined in floating point, and we should not extrapolate to contexts where that is not the case.
